Store files in ~/.local/share/mcfly/ instead of ~/.mcfly/ to avoid polluting
HOME directory.

TODO: Propagate to upstream.

--- a/src/settings.rs
+++ b/src/settings.rs
@@ -448,7 +448,9 @@
     pub fn storage_dir_path() -> PathBuf {
         home_dir()
             .unwrap_or_else(|| panic!("McFly error: Unable to access home directory"))
-            .join(PathBuf::from(".mcfly"))
+            .join(PathBuf::from(".local"))
+            .join(PathBuf::from("share"))
+            .join(PathBuf::from("mcfly"))
     }
 
     pub fn mcfly_db_path() -> PathBuf {
--- a/src/history/history.rs
+++ b/src/history/history.rs
@@ -746,11 +746,11 @@
         let commands =
             shell_history::full_history(&shell_history::history_file_path(), history_format);
 
-        // Make ~/.mcfly
+        // Make ~/.local/share/mcfly
         fs::create_dir_all(Settings::storage_dir_path())
             .unwrap_or_else(|_| panic!("Unable to create {:?}", Settings::storage_dir_path()));
 
-        // Make ~/.mcfly/history.db
+        // Make ~/.local/share/mcfly/history.db
         let connection = Connection::open(Settings::mcfly_db_path()).unwrap_or_else(|_| {
             panic!(
                 "Unable to create history DB at {:?}",
